### PR TITLE
add hw2

### PR DESCRIPTION
**Pull request by akvise**

### CODE REVIEW

File: lesson_2/.idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Datasource local storage ignored files
+/dataSources/
+/dataSources.local.xml
+# Editor-based HTTP Client requests
+/httpRequests/
```
Comment (dimonl) [inline lesson_2/.idea/.gitignore]: add this file in .gitignore and remove from PR

File: lesson_2/.idea/lesson_2.iml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="WEB_MODULE" version="4">
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (dimonl) [inline lesson_2/.idea/lesson_2.iml]: add this file in .gitignore and remove from PR

File: lesson_2/.idea/modules.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/lesson_2.iml" filepath="$PROJECT_DIR$/.idea/lesson_2.iml" />
+    </modules>
+  </component>
+</project>
```
Comment (dimonl) [inline lesson_2/.idea/modules.xml]: All this folder .idea should stay off the PR

File: lesson_2/pkg/fib/fib.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import "fmt"
+
+// NumL : count Fibonacci number by loop and print
+func NumL(n int) {
+	cur, prev :=  1, 0
+	for i := 0; i < n - 1; i++ {
+		prev, cur = cur, cur + prev
+	}
+	fmt.Println(cur)
+}
```
Comment (dimonl) [inline lesson_2/pkg/fib/fib.go]: function shouldn't print result. You should count and return result? to use it somewhere...

File: lesson_2/pkg/fib/fib.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+func rec(n int) int {
+	if n == 1 || n == 2{
+		return 1
+	}
+	return rec(n - 1) + rec(n - 2)
+}
+
+// NumR : print recursion func "rec" with Fib num
+func NumR(n int){
+	fmt.Println(rec(n))
```
Comment (dimonl) [inline lesson_2/pkg/fib/fib.go]: the same above. Try to return result and print it in main func

File: lesson_2/pkg/fib/fib.go [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package fib
+
+import "fmt"
+
+// NumL : count Fibonacci number by loop and print
+func NumL(n int) {
```
Comment (dimonl) [inline lesson_2/pkg/fib/fib.go]: strange naming

File: lesson_2/pkg/fib/fib.go [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+	cur, prev :=  1, 0
+	for i := 0; i < n - 1; i++ {
+		prev, cur = cur, cur + prev
+	}
+	fmt.Println(cur)
+}
+
+// rec : count Fibonacci number by recursion
+func rec(n int) int {
```
Comment (dimonl) [inline lesson_2/pkg/fib/fib.go]: strange naming

File: lesson_2/pkg/fib/fib.go [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+	if n == 1 || n == 2{
+		return 1
+	}
+	return rec(n - 1) + rec(n - 2)
+}
+
+// NumR : print recursion func "rec" with Fib num
+func NumR(n int){
```
Comment (dimonl) [inline lesson_2/pkg/fib/fib.go]: strange naming

File: lesson_2/cmd/app/main.go [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+func main() {
+	var n int
+
+	defer fmt.Println("Complete!")
+	fmt.Print("This program can count fibonacci numbers\n", "Please enter number: ")
+	fmt.Scan(&n)
+
+	fib.NumL(n) //by loop
+	fib.NumR(n)	//by recursion
```
Comment (dimonl) [inline lesson_2/cmd/app/main.go]: get here results from functions and then print it. It helps you to use package "fmt" only in one place(module) of your project

File: lesson_2/cmd/app/main.go [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+	var n int
+
+	defer fmt.Println("\nCompleted!")
+	fmt.Print("This program can count fibonacci numbers\n", "Please enter number: ")
+	fmt.Scan(&n)
+
+
+	fmt.Println("Count by with liner execution time:")
+	for i := 1; i < n; i++ {
+		fmt.Print(fibonacci.FibLoop(i)," ") 	//by loop
```
Comment (rat6666) [inline lesson_2/cmd/app/main.go:18]: fibonacci.FibLoop() Do we relly need Fib in function name? Seems Loop to be enough.

File: lesson_2/cmd/app/main.go [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+	}
+
+	fmt.Println("\nCount by recursion with exponential execution time:")
+	for i := 1; i < n; i++ {
+		fmt.Print(fibonacci.FibRecursion(i), " ") //by recursion
```
Comment (rat6666) [inline lesson_2/cmd/app/main.go:23]: the same here

**Review by rat6666 — APPROVED** [review]

**Review by dimonl — APPROVED** [review]

**Pull request merged**